extract-patches.sh

```diff
@@ -2,6 +2,13 @@
 set -euo pipefail
 IFS=$'\n\t'
 
+getIndexedSubmoduleSha() {
+    (
+        cd "$DIR"
+        git ls-tree HEAD "src/$REPO_NAME" | awk '{print $3}'
```

Comment:
I think this will fail for NuGet's nested submodules - `REPO_NAME` will just be `Common`, etc.

Reply:
Yep, fixed. I don't think we've done nested submodule patches in the build, but no need to prevent that here.